statistics/handle/statscache.go

```diff
@@ -39,7 +40,7 @@ type cacheItem struct {
 	cost  int64
 }
 
-func newStatsCache() statsCache {
+func newStatsCache(capacity int64) statsCache {
```

Comment:
What about using variable.StatsCacheMemQuota directly when needed instead of passing domain->handle->statscache, since it's already a global atomic.